Remove startup byte[] allocation in EventSource

stephentoub opened the pull request:
Only occurs once and is then cached, but it's hit on startup, and may as well remove it.

```
	IL_0000: ldsflda valuetype '<PrivateImplementationDetails>'/'__StaticArrayInitTypeSize=16' '<PrivateImplementationDetails>'::BABB0FF3A59F6738CDF14170DA56230D6528DC9D619C5860288B86A4E3B29D63
	IL_0005: ldc.i4.s 16
	IL_0007: newobj instance void valuetype System.ReadOnlySpan`1<uint8>::.ctor(void*, int32)
	IL_000c: stloc.0
```

cc: @noahfalk

AaronRobinsonMSFT commented on src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs:
```diff
@@ -3882,10 +3880,6 @@ private bool SelfDescribingEvents
         // creation of yet another EventSource.  When this happens, these statics may not yet be initialized.
         // Rather than depending on initialized statics, use lazy initialization to ensure that the statics are initialized exactly when they are needed.
-
-        // used for generating GUID from eventsource name
```
Does it make sense to retain this comment on the local variable above?

noahfalk replied:
IMO I'd leave the PR how it is now, but I am fine either way.

noahfalk reviewed (approved):
LGTM, thanks Stephen!

stephentoub committed:
Remove startup byte[] allocation in EventSource

Only occurs once and is then cached, but it's hit on startup, and may as well remove it.

stephentoub commented:
I had to put the old code back under ifdef to not break the downlevel assembly.  What is that library for?

noahfalk commented:
My understanding is that ES_BUILD_STANDALONE define is to build [this nuget package](https://www.nuget.org/packages/Microsoft.Diagnostics.Tracing.EventSource), which was offered as an out-of-band release vehicle for some EventSource functionality before it was available in the official runtime release. At this point I don't think we release this thing any more and I doubt there is any purpose to keeping the define or the [project](https://github.com/dotnet/runtime/blob/df3930ecd237000813d1833286513d65557efffe/src/libraries/Microsoft.Diagnostics.Tracing.EventSource.Redist/src/Microsoft.Diagnostics.Tracing.EventSource.Redist.csproj) in our source. 

@brianrob - does that sound right?

stephentoub commented:
We're still building https://github.com/dotnet/runtime/tree/master/src/libraries/Microsoft.Diagnostics.Tracing.EventSource.Redist. If we can remove that, great :)

brianrob commented:
It's actually this package: https://www.nuget.org/packages/Microsoft.Diagnostics.Tracing.EventSource.Redist/

You're right that it's an out-of-band release vehicle for EventSource - specifically for desktop, since it's hard to get fixes to desktop customers in a timely manner.  I would recommend not getting rid of this, as the problem of shipping fixes to desktop customers isn't likely to go away, and this is a thing that comes up every once and while, and is always a high-priority thing when it comes up.  The last time it was a big deal, we didn't have a way to produce a package, and thus I ended up doing the work to add it here.
